File: 768-partition-labels/partition-labels.js
var partitionLabels = function(s) {
    let lookup = {}

    for (let i = 0; i < s.length; i++) {
        lookup[s[i]] = i
    }

    let size = 0
    let end = 0
    let output = []
    for (let i = 0; i < s.length; i++) {
        size += 1
        end = Math.max(end, lookup[s[i]])
        if (i == end) {
            output.push(size)
            size = 0
        }
    }

    return output
};

File: 768-partition-labels/partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        lookup = {}
        output = []
        size = 0
        end = 0

        for i in range(len(s)):
            lookup[s[i]] = i

        for i in range(len(s)):
            size += 1
            end = max(end, lookup[s[i]])

            if (i == end):
                output.append(size)
                size =0
        return output
             
        